Mage/src/main/java/mage/abilities/dynamicvalue/common/PermanentsOnBattlefieldCount.java

```diff
@@ -13,8 +12,8 @@
  */
 public class PermanentsOnBattlefieldCount implements DynamicValue {
 
-    private FilterPermanent filter;
-    private Integer multiplier;
+    private final FilterPermanent filter;
+    private final Integer multiplier;
```

Comment:
Why does this need a Integer rather than an int?

Reply:
Because it can be null, which has the special meaning of "use a multiplier of 1, but generate text suitable for an "X, where X is..." template rather than a "{multiplier} for each..." template.